src/integrations/prefect-slack/prefect_slack/credentials.py

```diff
@@ -117,3 +122,20 @@ async def notify(self, body: str, subject: Optional[str] = None):
 
             if response.status_code >= 400:
                 raise NotificationError(f"Failed to send message: {response.body}")
+
+    async def notify_async(self, body: str, subject: Optional[str] = None):
+        """
+        Sends a message to the Slack channel.
+        """
+        await _notify_async(self, body, subject)
+
+    @async_dispatch(_notify_async)  # type: ignore
```

Comment:
Would `@async_dispatch(SlackWebhook.notify_async)` work here? I'm guessing it might since you're passing `self` through to `_notify_async` (assuming I understand this correctly).

Reply:
it wouldn't quite work, because `SlackWebhook` is not defined there

Reply:
Ah yes, of course